### PR TITLE
refactor server middlewares

### PR DESCRIPTION
**Pull request by dikaeinstein**
#### What does this PR do?
Refactor server middlewares
#### Description of Task to be completed?
Remove unnecessary if statements, reduce code complexity,
and remove code smell.
#### How should this be manually tested?
N/A
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories? (if applicable)
[Refactor validator middlewares](https://www.pivotaltracker.com/story/show/158415207)

### CODE REVIEW

File: server/middleware/menuValidator.js [hunk 1]
```diff
@@ -1,150 +1,104 @@
...
+
+        if (invalidMealIds.length > 0) {
+          throw new Error('mealIds can only be integer values');
+        }
+        return true;
+      });
+  }
+  return body('mealIds')
+    .exists().withMessage('Menu must have at least one meal')
+    .isArray().withMessage('Menu must have at least one meal')
```
Comment (bekomay26) [inline server/middleware/menuValidator.js:65]: Good work on the validator middleware but, you can make the message 'Menu must have at least one meal' more specific, i.e in context with what the method is validating. It is validating the datatype, not the data length.

Comment (dikaeinstein) [reply server/middleware/menuValidator.js:65]: Thank you for the feedback. I will implement your feedback in the code. I will also add more descriptive error messages that best describes an error in my code.

File: server/middleware/numberValidator.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { body } from 'express-validator/check';
+
+/**
+ * Validate meal ${fieldName}
+ *
+ * @param {Object} options - Options object with key optional
+ *
+ * @returns {ValidationChain} - ValidationChain middleware
+ */
```
Comment (Nkemjiks) [inline server/middleware/numberValidator.js:9]: Why is there no description of what this code does? I can barely understand what is going on here. This is not a clean code. You won't last long here.

Comment (dikaeinstein) [reply server/middleware/numberValidator.js:9]: I will add a better description to jsdoc comments to add clarity about what the function does

Comment (bekomay26) [reply server/middleware/numberValidator.js:9]: @Nkemjiks LOL

**Pull request merged**